src/java.base/share/classes/javax/crypto/CryptoPolicyParser.java

```diff
@@ -616,33 +617,27 @@ private static class CryptoPermissionEntry {
          * Calculates a hash code value for the object.  Objects
          * which are equal will also have the same hashcode.
          */
+        @Override
         public int hashCode() {
             int retval = cryptoPermission.hashCode();
-            if (alg != null) retval ^= alg.hashCode();
-            if (exemptionMechanism != null) {
-                retval ^= exemptionMechanism.hashCode();
-            }
+            retval ^= Objects.hashCode(alg);
+            retval ^= Objects.hashCode(exemptionMechanism);
             retval ^= maxKeySize;
-            if (checkParam) retval ^= 100;
-            if (algParamSpec != null) {
-                retval ^= algParamSpec.hashCode();
-            }
+            retval ^= (checkParam ? 100 : 0);
+            retval ^= Objects.hashCode(algParamSpec);
```

Comment:
After this change, someone will wonder why the code does all the intermediate assignments.

```
    return Objects.hashCode(alg)
        ^ Objects.hashCode(exemptionMechanism) 
        ^ (checkParam ? 100 : 0)  
        ^ Objects.hashCode(algParamSpec);
}

Here and elsewhere...